velox/functions/lib/Slice.cpp

```diff
@@ -77,12 +85,15 @@ class SliceFunction : public exec::VectorFunction {
       BaseVector::flattenVector(args[0]);
     }
 
-    VectorPtr localResult =
-        applyArray<int64_t>(rows, args, context, outputType);
+    VectorPtr localResult = kind_ == TypeKind::INTEGER
+        ? applyArray<int32_t>(rows, args, context, outputType)
+        : applyArray<int64_t>(rows, args, context, outputType);
     context.moveOrCopyResult(localResult, rows, result);
   }
 
  private:
+  // The type kind of start and length.
+  TypeKind kind_;
```

Comment:
I wonder if it's better to add kind as a template parameter.

Reply:
From mbasmanova's advice, he thinks template here is not necessary  https://github.com/facebookincubator/velox/pull/9620#discussion_r1617546384

Reply:
Makes sense to me. Thanks.